service/history/workflow/context.go

```diff
@@ -850,6 +849,9 @@ func (c *ContextImpl) ReapplyEvents(
 }
 
 func (c *ContextImpl) UpdateRegistry() update.Registry {
+	if c.updateRegistry == nil {
+		c.updateRegistry = update.NewRegistry(c.MutableState)
```

Comment:
We only assign to this value once. However, we assign to`c.MutableState` in 2 other places. Additionally, it's an exported field, so anyone can modify it. This introduces the possibility for the MS and the update registry to be out of sync. How about hiding this field, adding a getter for it, and then only assigning to `updateRegistry` in `LoadMutableState`?